src/interface.rs

```diff
@@ -57,11 +57,15 @@ where
     ///
     /// Enables direct interaction with the device with the help of [command()](Epd4in2::command())
     pub(crate) fn data(&mut self, spi: &mut SPI, data: &[u8]) -> Result<(), SPI::Error> {
-        // high for data
-        let _ = self.dc.set_high();
+        for val in data.iter().copied() {
+            // high for data
+            let _ = self.dc.set_high();
```

Comment:
There shouldn't be a need to set dc high each time, since the output pin for dc stays on the same level if we don't change anything.

Reply:
Good call - I pushed a change for data and data_x_times which only sets dc high once.